Automattic/newspack-plugin
# feat(ia): Settings / Connections - Plugins & Analytics

### All Submissions:


### Changes proposed in this Pull Request:

* Migrated from _/wp-admin/admin.php?page=newspack-connections-wizard_.
* Adds Plugin and Analytics sections/components.
* Refactored API Fetch to use new `useWizardApiFetch` hook, introduced in #3163.
* Refactored `ActionCard` components to use `WizardActionCard` & `WizardPluginCard`.
* Added type definitions to `useObjectState` to appease TypeScript.
* Includes major refactor introduced in #3188 

### How to test the changes in this Pull Request:

**Original locations:**
* Plugins: _/wp-admin/admin.php?page=newspack-connections-wizard_
* Analytics: _/wp-admin/admin.php?page=newspack-analytics-wizard_

1. Checkout `feat/ia-settings-connections__plugins` and build assets.
2. Navigate to _/wp-admin/admin.php?page=newspack-settings_

**Test 1:**
1. Wait for relevant sections to load.
2. Observe plugins and analytics load as expected. Can be compared to legacy instance of Plugins and Analytics sections - _/wp-admin/admin.php?page=newspack-connections-wizard_.

**Test 2:**
1. Run `npm run watch`.
2. Navigate to `assets/wizards/newspack/views/settings/connections/plugins.tsx`.
3. Enforce an error: Update `path` `'/newspack/v1/plugins/jetpack'` with `'/newspack/v1/plugins/jetpacks'`. Update `slug` `jetpack` with `jetpacks`.
4. Refresh browser and confirm error is present.
![Screenshot 2024-06-28 at 11 21 06](https://github.com/Automattic/newspack-plugin/assets/3676975/14fc7986-1e73-4293-8f98-81e56411b8b5)

**Test 3: (X represents the plugins name)**
1. Pick a plugin and make sure its uninstalled.
2. When observing this plugin within the context of this PR. Observe that there is a "Install X" call to action present.
3. Click `Install X`, wait for completion, then, confirm the plugin installs correctly and the CTA has changed to "Activate X"
4. Click "Activate X", wait for completion, then confirm the page reloads.
5. Now observe that the plugin either has a green check (signaling all is okay and no further action is required) or a "Complete Setup" CTA is present.
6. If "Complete Setup" CTA is shown click it and confirm it takes you to the relevant settings page.

**Additional steps:**
- Confirm data (including errors) are persistent between tab changes.

### Other information:
